source/common/config/subscription_factory_impl.cc

```diff
@@ -43,11 +43,6 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
                                                             api_config_source);
     const auto transport_api_version = Utility::getAndCheckTransportVersion(api_config_source);
     switch (api_config_source.api_type()) {
-    case envoy::config::core::v3::ApiConfigSource::hidden_envoy_deprecated_UNSUPPORTED_REST_LEGACY:
```

Comment:
Since the PGV annotations for the v3 version of this field `DEPRECATED_AND_UNAVAILABLE_DO_NOT_USE` don't force rejection in some situations, should we keep this? When we get rid of `generated_api_version` we can then either use a PGV annotation or the `DEPRECATED_AND_UNAVAILABLE_DO_NOT_USE` name here instead.